### DISCUSSION TITLE
(Newbie) Help converting DV profile 7 to 8 in a MKV container

### DISCUSSION THREAD

**Discussion by MrDover27** [quietvoid/dovi_tool#195] [Q&A] [answered]
I am completely new to this and I just want to convert a MKV file with DV profile 7 to 8 so DV can be triggered on my TV. I have tried looking at other discussions but Im not fully understanding my first few steps. How do I go about extracting the enhancement layer from profile 7, converting it profile 8 and then put it back into the base layer so the MKV reads as Dolby Vision profile 8? I think i'm understanding the steps on how to do this but not how to perform those steps.

**Comment by quietvoid** [comment, marked as answer]
The simplest way is to use `convert` and to remux the resulting HEVC file.  
As the readme instructions show:
`ffmpeg -i input.mkv -c:v copy -vbsf hevc_mp4toannexb -f hevc - | dovi_tool -m 2 convert --discard -`

This converts the RPU to profile 8, and discards the enhancement layer, leaving you with a base layer + RPU track.

And then remux the new video with the rest of the tracks.

**Reply by MrDover27** [reply]
Im getting the error "At least one output file must be specified". Where should put an output file in that line of code?

**Reply by quietvoid** [reply]
Sounds like you didn't copy the line properly..
That error is coming from ffmpeg.

You need to keep the trailing `-` so that it pipes the stream to `dovi_tool`.

**Reply by MrDover27** [reply]
I highlighted and copied it exactly how it is but with my input file. Is there a way to do this as a separate action in ffmpeg and then dovi_tool afterwards instead of one single command line? Maybe it will work for me that way?

**Reply by quietvoid** [reply]
Yes.

```console
ffmpeg -i input.mkv -c:v copy video.hevc
dovi_tool -m 2 convert --discard video.hevc
```

**Reply by sw5163** [reply]
Hi, quietvoid,
Does the default mode 2 support `remove mapping` now?
I thought an additional json config is required for `profile 7 fel` -> `profile 8.1`?

**Reply by quietvoid** [reply]
> Hi, quietvoid, Does the default mode 2 support `remove mapping` now? I thought an additional json config is required for `profile 7 fel` -> `profile 8.1`?

No, mode 2 doesn't do `remove_mapping`.  
And I'm not sure I'd want it to, even for FEL only.

**Reply by sw5163** [reply]
Is there any official document about how exactly those mapping data work?

Currently I thought P5 RPU has `reshaping data` for `ICtCp`->`YCbCr`.
P8.1 RPU doesn't need `reshaping data` for `HDR10`->`HDR10`.
P8.4 RPU has `reshaping data` for `HLG`->`PQ`.
P7 RPU has `reshaping data` for processing the weird colour in EL.

But is there an expression or a formula?
Please correct me if I'm wrong, thanks!

**Reply by MrDover27** [reply]
> The simplest way is to use `convert` and to remux the resulting HEVC file. As the readme instructions show: `ffmpeg -i input.mkv -c:v copy -vbsf hevc_mp4toannexb -f hevc - | dovi_tool -m 2 convert --discard -`
> 
> This converts the RPU to profile 8, and discards the enhancement layer, leaving you with a base layer + RPU track.
> 
> And then remux the new video with the rest of the tracks.

I figured it out from doing a little research into the ffmpeg error "At least one output file must be specified". All I needed to do was add quotes around the input file and it works. So it would just be 

`ffmpeg -i "input.mkv" -c:v copy -vbsf hevc_mp4toannexb -f hevc - | dovi_tool -m 2 convert --discard -`
